Chat view padding issue removed

muhammadusama95 opened the pull request:
### Details
Implemented Solution is:
Go to ReportActionCompose.js File. Go to Parent View then there are multiple containers i.e. attachments, send button, and input text field. There was a problem with the input text field view styles. There wad Unnecessary padding of the input text field which was causing the extra space at the top and was not setting the input text view to the center of the parent view So we removed that and input text is at the center of the parent view and the top spacing has been removed using top, bottom padding value to 0 and I have tested this.
I am waiting to add me to the git link SO I can push the code to my branch and your team can review the code.


### Fixed Issues
Remove Top Padding from chatView.

### Tests
Tested on Android OnePlus 3 Mobile.
Tested on iOS iPhone X emulator.



### Tested On
- [x] iOS
- [x] Android

#### iOS
![iOS_ScreenShot](https://user-images.githubusercontent.com/49236138/103238893-e2c8b780-496d-11eb-8819-609c356ecfff.png)


#### Android
![android_screenshot](https://user-images.githubusercontent.com/49236138/103238891-e1978a80-496d-11eb-81eb-8eb646446631.jpeg)

14f8127 committed:
Updated Commit

14f8127 committed:
Top Space issue removed

roryabraham commented:
Hi @muhammadusama95 - just a few early comments on this PR:

1) Please include the link to the issue you've been assigned under the **Fixed Issue** header
2) Please be sure to test any and all changes on all platforms, including screenshots (web, desktop, mobile web, iOS, Android). Right now, I only see iOS and Android
3) Looking at the screenshot you provided on Android, the text in the input field is vertically misaligned

14f8127 committed:
Nothing Changed Just Added Comments.

muhammadusama95 commented:
Hi, @roryabraham Thanks for your review of my code.
I just made a new commit, Added Comments on code where I changed for the fixing of the top padding issue. So you can easily
identify my code.
Below are my answers to all your questions:
1: https://github.com/Expensify/Expensify.cash/issues/1002

2: Yeah As per my discussion with Michael, This is a cross-platform app So I have tested it on both iOS and Android. Not included web, desktop, or mobile web version.

3: By mistakenly added an old android screenshot Please see attached updated android screenshot too.
![UpdatedAndroidScreenshot](https://user-images.githubusercontent.com/49236138/103297891-05fd7080-4a1b-11eb-8761-a8bc40b3bce1.jpeg)

4: Now I added comments in the new commit So you can quickly identify my changes.

5: There is no new lint error or warnings in my branch as I just changed the values of padding.

roryabraham reviewed (changes requested):
Hi @muhammadusama95, sorry if I wasn't clear before, but I have a few new comments.

1) All the code style changes you have made to `src/styles/styles.js` are unacceptable, because they do not conform to our code style guidelines and the code format throughout the rest of the codebase. Adding comments to identify your code is not a valid solution - the comments would litter the codebase with irrelevant information. So to be clear, we will not merge any PR containing extraneous changes not relevant to the issue which you have been assigned. Why did you remove all indentation from the file?
2) _All_ PRs must be tested on 5 platforms: web, mobile web, desktop, android, and iOS. The same code is used to build the application on all 5 platforms, so we need to make sure your changes don't cause undesirable effects in the other platforms.
3) It seems like the screenshots you've provided render only the `ReportActionCompose` component in isolation, not the complete application.
4) I noticed that your code contains a platform-check, like so:

```
if (Platform.OS == 'ios') {
    ...
}
```

This is something we have decided to intentionally avoid in this codebase. Instead, check out the [README section covering platform-specific file extensions](https://github.com/Expensify/Expensify.cash#platform-specific-file-extensions).

14f8127 committed:
Issue Fixed with out platform check

14f8127 committed:
intentation fixed

14f8127 committed:
intentation and padding issue fixed without platform check and removed comments

muhammadusama95 commented:
Hi @roryabraham,
Please check the last 3 commits So you can check the indentation Set, Comments removed, and padding issue fixed.

Now I have tested it using login through Expensify app and check below screenshots of both iOS and Android respectively.
![iOS_ScreenShot_updated](https://user-images.githubusercontent.com/49236138/103379564-db3d1600-4b07-11eb-90f2-645d52c8f84c.png)
![Android_ScreenShot_updated](https://user-images.githubusercontent.com/49236138/103379567-dd06d980-4b07-11eb-8b18-8678f1a04587.png)

About testing it on All Platforms @michaelhaxhiu had chat with me on android and iOS only. Check Below Screenshot.
![Michael_ScreenShot](https://user-images.githubusercontent.com/49236138/103379680-4850ab80-4b08-11eb-850a-187787305665.PNG)

roryabraham commented:
Hi @muhammadusama95. Regardless of what others may have told you, _all PRs must be tested on all platforms_. So that means that your code changes must be tested on:

1) Web (from a laptop)
2) Desktop (from a mac computer - let me know if you don't have a mac)
3) Web (from a mobile device or simulator)
4) iOS
5) Android

roryabraham reviewed (changes requested):
The diff is more reasonable now, but please review the diff and remove all non-critical changes.

roryabraham commented on src/styles/styles.js:
```diff
@@ -8,7 +8,7 @@ import variables from './variables';
 
 const styles = {
-    // Utility classes
+// Utility classes
```
This indentation change is unnecessary, please revert

roryabraham commented on src/pages/home/report/ReportActionCompose.js:
```diff
@@ -175,7 +175,7 @@ class ReportActionCompose extends React.Component {
                                     multiline
                                     ref={el => this.textInput = el}
-                                    textAlignVertical="top"
+                                    
```
Please remove this blank line

roryabraham commented on src/styles/styles.js:
```diff
@@ -37,24 +37,17 @@ const styles = {
+    mt2: {
+        marginTop: 20,
+    },
+
```
Please remove this blank line

roryabraham commented on src/styles/styles.js:
```diff
@@ -37,24 +37,17 @@ const styles = {
-    mt2: {
-        marginTop: 8,
-    },
+
```
Please remove this blank line

roryabraham commented on src/styles/styles.js:
```diff
@@ -68,12 +61,7 @@ const styles = {
-    mb6: {
-        marginBottom: 24,
-    },
+
```
Please remove this blank line

roryabraham commented on src/styles/styles.js:
```diff
@@ -361,6 +327,7 @@ const styles = {
         paddingTop: 10,
         paddingBottom: 10,
         textAlignVertical: 'center',
+
```
Please remove this blank line

roryabraham commented on src/styles/styles.js:
```diff
@@ -339,7 +305,7 @@ const styles = {
     },
 
-    // Actions
+// Actions
```
This indentation change is unnecessary, please revert

roryabraham commented on src/styles/styles.js:
```diff
@@ -437,26 +382,13 @@ const styles = {
-    },
-
-    // Sidebar Styles
+// Sidebar Styles
```
Please indent this comment four spaces

roryabraham commented on src/styles/styles.js:
```diff
@@ -608,7 +536,7 @@ const styles = {
     },
 
-    // App Content Wrapper styles
+// App Content Wrapper styles
```
This indentation change is unnecessary, please revert

roryabraham commented on src/styles/styles.js:
```diff
@@ -676,7 +604,7 @@ const styles = {
     },
 
-    // Chat Item
+// Chat Item
```
This indentation change is unnecessary, please revert

roryabraham commented on src/styles/styles.js:
```diff
@@ -705,18 +633,16 @@ const styles = {
-        fontWeight: fontWeightBold,
+        height: 24,
         lineHeight: 20,
+        fontWeight: '600',
```
The change to `fontWeight` here shouldn't be necessary - please use `fontWeightBold` instead

roryabraham commented on src/styles/styles.js:
```diff
@@ -1060,11 +988,11 @@ const baseCodeTagStyles = {
-    // styles in the below "tagStyles" instead. If you need to reuse those
-    // styles from the renderer, just pass the "style" prop to the underlying
-    // component.
+// As of react-native-render-html v6, don't declare distinct styles for
```
This indentation change is unnecessary, please revert

roryabraham commented on src/styles/styles.js:
```diff
@@ -1090,7 +1021,7 @@ const webViewStyles = {
             marginBottom: 4,
 
-            // Overwrite default HTML margin for blockquotes
+// Overwrite default HTML margin for blockquotes
```
This indentation change is unnecessary, please revert

muhammadusama95 commented:
Hi @roryabraham,
Please find All Platform tested screenshots.
1: Android
![Android_ScreenShot_updated](https://user-images.githubusercontent.com/49236138/103442631-caacad00-4c79-11eb-9ebf-553da8adada0.png)
2:iOS
![iOS_ScreenShot_updated](https://user-images.githubusercontent.com/49236138/103442630-c3859f00-4c79-11eb-8d47-adb811c1d2dd.png)
3:Mobile Web
![mobile_web](https://user-images.githubusercontent.com/49236138/103442606-a0f38600-4c79-11eb-8c7e-036e6be414f3.png)
4:Web
![web_](https://user-images.githubusercontent.com/49236138/103442616-acdf4800-4c79-11eb-9488-5be127d22334.png)
5:MAC-Desktop
![mac_desktop](https://user-images.githubusercontent.com/49236138/103442614-abae1b00-4c79-11eb-8496-42377e897b27.png)

14f8127 committed:
indentation revert

roryabraham commented:
Looks like you've got a merge conflict

roryabraham commented on src/pages/home/report/ReportActionCompose.js:
```diff
@@ -218,7 +217,7 @@ class ReportActionCompose extends React.Component {
                         />
                     </TouchableOpacity>
                 </View>
-                <ReportTypingIndicator reportID={this.props.reportID} />
```
Please remove this unnecessary whitespace change

roryabraham commented on src/styles/styles.js:
```diff
@@ -37,24 +37,15 @@ const styles = {
         marginLeft: 8,
     },
 
+    mt2: {
```
Please don't change the indentation utils here. If you need `marginTop: 20`, use `mt5`

roryabraham replied:
Also please make sure that these are arranged sequentially like `mt1, mt2, mt3...`

roryabraham commented on src/styles/styles.js:
```diff
@@ -37,24 +37,15 @@ const styles = {
     mt3: {
         marginTop: 12,
     },
-    mt4: {
```
Why did you remove these three styles?

roryabraham commented on src/styles/styles.js:
```diff
@@ -68,12 +59,7 @@ const styles = {
     mb4: {
         marginBottom: 16,
     },
-    mb5: {
```
Why did you remove these two styles?

roryabraham commented on src/styles/styles.js:
```diff
@@ -95,9 +81,6 @@ const styles = {
         width: '100%',
         height: '100%',
     },
-    width50p: {
```
Why did you remove this style?

roryabraham commented on src/styles/styles.js:
```diff
@@ -201,7 +166,6 @@ const styles = {
 
     textP: {
         color: themeColors.text,
-        fontFamily: fontFamily.GTA,
```
Why did you remove the fontFamily?

roryabraham commented on src/styles/styles.js:
```diff
@@ -175,24 +158,6 @@ const styles = {
         height: '100%',
     },
 
-    link: {
```
Why did you remove these styles?

roryabraham commented on src/styles/styles.js:
```diff
@@ -1060,11 +986,11 @@ const baseCodeTagStyles = {
-    // styles in the below "tagStyles" instead. If you need to reuse those
-    // styles from the renderer, just pass the "style" prop to the underlying
-    // component.
+   // As of react-native-render-html v6, don't declare distinct styles for
```
These comments need to be indented by one space

michaelhaxhiu commented:
Let's give @roryabraham a chance to review, as he's been doing 🔧

roryabraham commented:
@muhammadusama95 There are still lots of unresolved comments that would need to be addressed before this could be merged. Not to mention there's a merge conflict, so it's not even possible to merge presently.

muhammadusama95 commented:
@michaelhaxhiu @roryabraham Can you please assign this issue to @ahmad.zahid2000@gmail.com. I will create a new branch and take a pull request instead of resolving these comments by doing this it's hardly 20-25 minutes work.

muhammadusama95 commented:
On the new branch, you will get only the required changes. Nothing else.

michaelhaxhiu commented:
@muhammadusama95 We have decided to move forward with another contributor. Thanks for your assistance thus far, but no further action is required from your side.

I'm closing this PR and going to pay you in Upwork shortly.
